util/masking.h

```diff
@@ -38,15 +38,12 @@ namespace dials { namespace util {
         : resolution_(
           af::c_grid<2>(panel.get_image_size()[1], panel.get_image_size()[0])) {
       vec3<double> s0 = beam.get_s0();
+      double wavelength_r = 1.0 / beam.get_wavelength();
       for (std::size_t j = 0; j < resolution_.accessor()[0]; ++j) {
         for (std::size_t i = 0; i < resolution_.accessor()[1]; ++i) {
           vec2<double> px(i + 0.5, j + 0.5);
-          try {
-            resolution_(j, i) = panel.get_resolution_at_pixel(s0, px);
-          } catch (dxtbx::error) {
-            // Known failure: resolution at beam center is undefined
-            resolution_(j, i) = 0.0;
-          }
+          vec3<double> s1 = panel.get_pixel_lab_coord(px).normalize() * wavelength_r;
```

Comment:
Wonder if any benefit in manually computing lab frame coordinate rather than calling `get_pixel_lab_coord` - would mean compiler could see what you are doing, may unroll loops in clever way etc.

Reply:
Possibly, however I think this change is a demonstrable enough improvement that it is worth including as is. We have discussed elsewhere other possible ways for computing the resolution map which may be better still.